### PR TITLE
Make dev profile faster by compiling deps with opt-level=3

### PR DESCRIPTION
**Pull request by tomaka**
As the code explains it, this is a list of dependencies that we compile with `opt-level=3` when doing a plain `cargo build`.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -180,6 +180,130 @@ members = [
...
+# a dependency on Substrate, you have to copy-paste this list in your own `Cargo.toml` (assuming
+# that you want the same list). This list is only relevant when running `cargo build` from within
+# the Substrate workspace.
+#
+# If you see an error mentioning "profile package spec ... did not match any packages", it
+# probably concerns this list.
+#
+# This list is ordered alphabetically.
+[profile.dev.package.aes-ctr]
+opt-level = 3
```
Comment (bkchr) [inline Cargo.toml:199]:
```suggestion
[profile.dev.package]
aes-ctr = { opt-level = 3 }
```

Doesn't this work as well?

Comment (gnunicorn) [inline Cargo.toml:199]: Any reason you've chosen that representation over
```suggestion
[profile.dev.package]
aes-ctr = { opt-level = 3 }
aes-soft = { opt-level = 3 }
```

To me this is more legible...

**Comment by tomaka** [comment]
Oh... well, I'll re-do the PR.

**Comment by tomaka** [comment]
For reference, this seems to cause warnings to be printed when someone runs `cargo install --git .. subkey`.
I've opened https://github.com/rust-lang/cargo/issues/8328

**Pull request merged**